app/controllers/notifications_controller.rb

```diff
@@ -66,4 +67,12 @@ def if_not_signed_in
         end
       end
     end
+
+    # Stops here if the signed in user is not the owner of
+    # the notification set by :set_notification.
+    def ensure_user_owns_notification!
+      unless @notification.user == current_user
```

Comment:
I like your suggestion @tcdowney 💯 . Combining the new before action with the existing set_notification will save us from problems in the future. 

@esBeee Definitely open an issue for a global method to check whether a user is an owner of the object. Great suggestion!